packages/apputils/src/dialog.tsx

```diff
@@ -436,12 +436,14 @@ export class Dialog<T> extends Widget {
         event.preventDefault();
 
         const activeEl = document.activeElement;
-        let index: number | undefined;
+        let index: number | undefined = this._defaultButton;
 
         if (activeEl instanceof HTMLButtonElement) {
           index = this._buttonNodes.indexOf(activeEl);
+          this.resolve(index !== -1 ? index : 0);
```

Comment:
Thinking about why index might be `-1`. This could be if the react body had a button which is not a button node. I think in that case we would not want to resolve either. Say if there was "Help" button in extension author's code.

Reply:
I will try that and see what tests on CI think:

```suggestion
          if (index !== -1) {
            this.resolve(index);
          }
```